Data/Aeson/Types/ToJSON.hs

```diff
@@ -515,6 +529,28 @@ contramapToJSONKeyFunction h x = case x of
     ToJSONKeyText  f g -> ToJSONKeyText (f . h) (g . h)
     ToJSONKeyValue f g -> ToJSONKeyValue (f . h) (g . h)
 
+-- 'toJSONKey' for 'Generic' types.
+-- These types must be sums of nullary constructors, whose names will be used
```

Comment:
Deriving is supported for enumeration types, i.e. the sums of nullary constructors. The names of constructors will be used as keys for JSON objects.